### PR TITLE
Extend parameter values building block

### LINKED ISSUE #1245: Extend Parameter Values Building Block
Refer to discussion #1226

This is to separate the two issues itemized into separate tasks.

> "Extend" action for the Parameter Values BB (maybe rename to "Add local molecule parameters"?):
Open a dialog for selection of a spatial structure (from one of the modules) and multi-selection of molecules (same view as for "Extend" of Initial Conditions?):
image

> Adds entries for all local molecule parameters in all physical containers (same logic as for "extend" of IC BB) that are described by a constant formula.

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #1245 

# Description
Allow users to add parameter values from selected spatial structure and molecules

## Type of change

Please mark relevant options with an `x` in the brackets.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: src/MoBi.UI/Views/BasePathAndValueEntityView.cs [hunk 1]
```diff
@@ -232,10 +234,28 @@ private void initColumnVisibility()
...
+               initializePathColumnVisibleIndex(i);
          }
       }
 
+      // When hiding and showing columns automatically based on presence of path elements
+      // we have to override how the visibility index is set. That's because visibility index
+      // of a column is updated as other columns are hidden. So, if you iterate in order and hide all columns
+      // then their internal hidden visibility index will be 0. If you iterate in order and show them,
+      // they appear in the reverse order
+      private void initializePathColumnVisibleIndex(int i)
```
Comment (rwmcintosh) [inline src/MoBi.UI/Views/BasePathAndValueEntityView.cs:251]: I found that when an empty parameter values editor suddenly gets parameter values the column order is messed up. It's the same with Initial Conditions.

If you closed and opened the editor, the path column order would be fixed.

File: src/MoBi.UI/Views/ParameterValuesView.cs [hunk 2]
```diff
@@ -39,16 +39,12 @@ public void AttachPresenter(IParameterValuesPresenter presenter)
 
       protected override void DoInitializeBinding()
       {
+         base.DoInitializeBinding();
+
          _unitControl.ParameterUnitSet += setParameterUnit;
 
          _dimensionComboBoxRepository.FillComboBoxRepositoryWith(_dimensionFactory.DimensionsSortedByName);
 
-         var colName = _gridViewBinder.AutoBind(dto => dto.Name)
```
Comment (rwmcintosh) [inline src/MoBi.UI/Views/ParameterValuesView.cs:46]: pulled up

**Review by msevestre — APPROVED** [review]
looks good to me. One suggestion for ToList to be removed that feels out of place

**Comment by rwmcintosh** [comment]
> looks good to me. One suggestion for ToList to be removed that feels out of place

Removing those in my next PR which is an extension to this one.

**Pull request merged**